lib/libgtkwave/src/gw-ghw-loader.c

```diff
@@ -315,12 +316,38 @@ static void recurse_tree_fix_from_whichcache(GwGhwLoader *self, GwTreeNode *t)
 
 static void incinerate_whichcache_tree(ghw_Tree *t)
 {
-    if (t->left)
-        incinerate_whichcache_tree(t->left);
-    if (t->right)
-        incinerate_whichcache_tree(t->right);
+    // Use a dynamic GArray to store the pending nodes. In some cases, the
+    // trees can be very deep, so we can't use explicit recursion here.
+    GArray *pending_nodes;
+    pending_nodes = g_array_new(FALSE, FALSE, sizeof (ghw_Tree*));
```

Comment:
It is better to use `GPtrArray` for an array of pointers, because the code is simplified and the type makes it obvious that it is an array of pointers:
```suggestion
    GPtrArray *pending_nodes = g_ptr_array_new();
```

You'll also need to adjust the other `g_array_...` calls to their matching `g_ptr_array_...` variant below.